crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -44,7 +45,7 @@ impl<'w> EntityRef<'w> {
     }
 
     #[inline]
-    pub fn world(&mut self) -> &World {
+    pub fn world(self) -> &'w World {
```

Comment:
Why does this take a `self` rather than `&self`?

Reply:
since its `Copy` theres no real reason to take `&self` It doesn't really make a difference

Reply:
Seems a bit weird to consume / take ownership here, even if Copy makes this behave normally. Unless we have codegen reasons to use this pattern, I think it creates fewer questions in the reader's mind if we use `&self`. And using copying here also involves "more" data transfer than passing in a `self` pointer. Probably doesn't matter, but that feels like another reason to bias toward `&self`.